dask/dataframe/io/parquet.py

```diff
@@ -174,11 +147,147 @@ def _read_parquet_row_group(open, fn, index, columns, rg, series, categories,
         return df
 
 
+# ----------------------------------------------------------------------
+# PyArrow interface
+
+def _read_pyarrow(fs, paths, file_opener, columns=None, filters=None,
+                  categories=None, index=None):
+    if not pyarrow_parquet:
+        raise ImportError("pyarrow.parquet not installed")
+
+    api = pyarrow_parquet
+
+    if filters is not None:
+        raise NotImplemented("Predicate pushdown not implemented")
+
+    if categories is not None:
+        raise NotImplemented("Categorical reads not yet implemented")
+
+    if isinstance(columns, tuple):
+        columns = list(columns)
+
+    columns = columns
+
+    dataset = api.ParquetDataset(paths)
+    schema = dataset.schema.to_arrow_schema()
+    task_name = 'read-parquet-' + tokenize(dataset, columns)
+
+    pieces = dataset.pieces
+
+    if columns is None:
+        all_columns = schema.names
+    else:
+        all_columns = columns
+
+    if not isinstance(all_columns, list):
+        out_type = Series
+        all_columns = [all_columns]
+    else:
+        out_type = DataFrame
+
+    divisions = (None,) * (len(dataset.pieces) + 1)
+
+    dtypes = _get_pyarrow_dtypes(schema)
+
+    meta = _meta_from_dtypes(all_columns, schema.names, dtypes)
+
+    task_plan = {
+        (task_name, i): (_read_arrow_parquet_piece,
+                         file_opener,
+                         piece, all_columns,
+                         out_type == Series,
+                         dataset.partitions)
```

Comment:
It's a small data structure containing the partition set; its size is proportional to the number of partitions. I haven't optimized its serialization time, but if it seems to be affecting things with large # of partitions:

```python
In [14]: p1 = pq.PartitionSet('foo', keys=['foo_{0}' for i in range(10000)])

In [15]: p2 = pq.PartitionSet('bar', keys=['bar{0}' for i in range(1000)])

In [16]: partitions = pq.ParquetPartitions()

In [17]: partitions.levels = [p1, p2]

In [18]: timeit pickle.dumps(partitions)
10000 loops, best of 3: 152 µs per loop
```